src/main/java/systems/crigges/jmpq3/JMpqEditor.java

```diff
@@ -209,25 +212,59 @@ private void readAttributesFile() {
         }
     }
 
+    /**
+     * For use when the MPQ is missing a (listfile)
+     * Adds this custom listfile into the MPQ and uses it
+     * for rebuilding purposes.
+     * If this is not a full listfile, the end result will be missing files.
+     *
+     * @param externalListfilePath  Path to a file containing listfile entries
+     */
+    public void setExternalListfile(File externalListfilePath) {
+        if(!externalListfilePath.exists()) {
+            log.warn("External MPQ File: " + externalListfilePath.getAbsolutePath() +
+                " does not exist and will not be used");
+            return;
+        }
+        try {
+            // Read and apply listfile
+            listFile = new Listfile(Files.readAllBytes(externalListfilePath.toPath()));
+            checkListfileEntries();
+            removeMissingFiles();
+            // Operation succeeded and added a listfile so we can now write properly.
+            // (as long as it wasn't read-only to begin with)
+            canWrite = !openedAsReadOnly;
+        } catch (Exception ex) {
+            log.warn("Could not apply external listfile: " + externalListfilePath.getAbsolutePath());
+            // The value of canWrite is not changed intentionally
+        }
+    }
+
+    /**
+     * Removes files from the listfile if they aren't
+     * actually in the map.
+     */
+    private void removeMissingFiles() {
```

Comment:
Seems to be equivalent to https://github.com/inwc3/JMPQ3/pull/36/files#diff-fb7d083bb9ea8eb0c5d21c0cc0059b5bR305 ?

Reply:
Yes, it is. Good point, that's a much better way of doing this.

Reply:
Is it resolved?